### ISSUE TITLE
Wrap strings automatically with dartfmt

### ISSUE THREAD

**Issue by kaspnilsson** [dart-lang/dart_style#909]
One of the best parts of Dart is the tooling, and the lack of comment and string wrapping (especially while those rules are enforced by presubmits) is a pretty major missing feature. AFAICT it should be reasonably fixable to wrap strings at 80ch (although I could easily be missing the complexity here)

**Comment by kaspnilsson** [comment]
Any chance we can get this looked at and prioritized? This is pretty obnoxious in practice to not have, and lots of other format tools have already solved for this problem

**Comment by munificent** [comment]
I agree this would be handy in many cases, but I think it would also do the *wrong* or at least confusing things in other cases. See #431 and https://github.com/dart-lang/dart_style/wiki/FAQ#why-does-the-formatter-only-touch-whitespace-by-default for some more context.

I'll leave this open mainly to so that others can find the issue instead of re-filing it, but I think it's unlikely that this will be feasible.

**Comment by kaspnilsson** [comment]
It's disappointing to hear that this is still the Dart team's POV. 

I would very, very strongly challenge your characterization of this problem as infeasible -- most other modern languages and toolings (Prettier.JS, gofmt, clang-format, etc) do this, at least in some form. Taking a look at the tooling that exists in other ecosystems would definitely expand your opinion as to what is feasible with a little bit of code.

I definitely can understand if this kind of work is not a priority right now, or if this particular tool is unstaffed or itself not a priority for the Dart team, etc, but I would really hope that this basic convenience is a feature of future formatters.

**Comment by munificent** [comment]
> most other modern languages and toolings (Prettier.JS, gofmt, clang-format, etc) do this, at least in some form.

* Prettier [does not and will not](https://github.com/prettier/prettier/issues/4123): "Closing this as it's definitely not going to happen in this project (Prettier core)."

* gofmt does not do line breaking *at all*.

* clang-format *does* have an option to support splitting strings. However, it does not appear to *unsplit* strings when they no longer need to be split. That violates one of the goals of dartfmt that its changes should generally be "reversible".

By reversible, I mean that if you:

0. Start with some code.
1. Change the code.
2. Run dartfmt.
3. Change the code back.
4. Run dartfmt again.

You should end up with the same code at 4 that you had at 0. If you make a string longer on step 1 and dartfmt splits it on 2, and then you make it shorter again on 3 but dartfmt doesn't *merge* the string back on 4, then your code is left with a pointless split that dartfmt inserted but you don't realize can be removed.
